pkg/test/docker.go

```diff
@@ -180,3 +195,46 @@ func (d *dockerRunner) RemoveAllContainers() {
 		}
 	}
 }
+
+func (d *dockerRunner) setHostMapping(resource *dockertest.Resource, mapping hostMapping) error {
+	timeout := time.Duration(1) * time.Millisecond
+	containerIp := resource.Container.NetworkSettings.Networks["bridge"].IPAddress
+	gatewayIp := resource.Container.NetworkSettings.Networks["bridge"].Gateway
+
+	addresses := []string{containerIp, gatewayIp, "127.0.0.1"}
```

Comment:
i think due to the nature of container port exposing, the containerIp will never be reachable under the dialPort unless it is the one-in-a-million case where traffic port and exposed port are actually the same.
If you want to reach the container ip with a given port, it needs to be the one that is exposed afaik. (say 8080 for localstack healthcheck)

Reply:
i would have expected our "CI" settable ip to be in here via os.GetEnv

Reply:
should we maybe add the mapping.setHost value into the slice too?

Reply:
Yes, it works, because the `dialPort` is a pointer, and before this code is executed the ports are updated. The container-port actually works like a charm on my mac.

Reply:
but you are executing the setPortMapping before your host mapping. Given you have a container that internally exposes port 8080 and your config has set port 0 (so auto-assign), then you already have set the settings.Port to the auto-assigned port and your mac is likely to use 127.0.0.1, no?